sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/LockRenewalOperation.java

```diff
@@ -214,8 +199,32 @@ public void close() {
                 logger.info("token[{}]. nextExpiration[{}]. next: [{}]. isSession[{}]", lockToken, offsetDateTime, next,
                     isSession);
 
-                sink.next(MessageUtils.adjustServerTimeout(next));
+                sink.next(calculateRenewalDelay(offsetDateTime));
                 return offsetDateTime;
             });
     }
+
+    private Duration calculateRenewalDelay(OffsetDateTime initialLockedUntil) {
+        final OffsetDateTime now = OffsetDateTime.now();
+        final Duration remainingTime = Duration.between(now, initialLockedUntil);
+
+        // If remaining time to renew is less than 400ms, renew immediately
+        // Lock renewal to follow the same behavior as C# SDK
+        // C# class : github.com/azure-sdk-for-net/.../Azure.Messaging.ServiceBus/src/Processor/ReceiverManager.cs#L367
+
+        if (remainingTime.isNegative() || remainingTime.toMillis() < 400) {
```

Comment:
Isn't checking `< 400 ms` sufficient?
```suggestion
        if (remainingTime.toMillis() < 400) {
```

Reply:
Negative is an interesting situation, but you are right makes sense, removed the reduandant check